lib/robots/dor_repo/accession/rights_metadata.rb

```diff
@@ -10,17 +10,34 @@ def initialize
         end
 
         def perform(druid)
-          obj = Dor.find(druid)
-          builder = DatastreamBuilder.new(datastream: obj.rightsMetadata)
+          object = DruidTools::Druid.new(druid, Dor::Config.stacks.local_workspace_root)
+          path = object.find_metadata('rightsMetadata.xml')
+          object_client = Dor::Services::Client.object(druid)
+          if path
+            object_client.metadata.legacy_update(
+              rights: {
+                updated: File.mtime(path),
+                content: File.read(path)
+              }
+            )
+          elsif !has_rights_metadata?(druid)
+            Honeybadger.notify("I don't think this ever happens because rights is created when registering. This is an experiment")
 
-          builder.build do |datastream|
-            build_datastream(obj, datastream)
+            # TODO: Fetch admin_policy_object without involving Fedora 3 concepts
+            apo = Dor.find(druid).admin_policy_object
+            object_client.metadata.legacy_update(
+              rights: {
+                updated: Time.now,
+                content: apo.defaultObjectRights.content
+              }
+            )
           end
```

Comment:
Is there an else case that needs to be handled? (rightsMetadata.xml not on disk and object has rights metadata)

Reply:
I don't think so, I think the two legitimate cases are "there is rightsMetadata.xml on disk" and "do nothing" (because registration gives every object rightsMetadata.xml).